src/AlcPacket.cpp

```diff
@@ -110,19 +117,29 @@ LibFlute::AlcPacket::AlcPacket(char* data, size_t len)
                         break; // ignored
                       }
       case EXT_FTI: {
-                      if (_fec_oti.encoding_id == FecScheme::CompactNoCode) {
-                        if (hel != 4) {
-                          throw "Invalid length for EXT_FTI header extension";
-                        }
-                        _fec_oti.transfer_length = (uint64_t)(ntohs(*(uint16_t*)hdr_ptr)) << 32;
-                        hdr_ptr += 2;
-                        _fec_oti.transfer_length |= (uint64_t)(ntohl(*(uint32_t*)hdr_ptr));
-                        hdr_ptr += 4;
-                        hdr_ptr += 2; // reserved
-                        _fec_oti.encoding_symbol_length = ntohs(*(uint16_t*)hdr_ptr);
-                        hdr_ptr += 2;
-                        _fec_oti.max_source_block_length = ntohl(*(uint32_t*)hdr_ptr);
-                        hdr_ptr += 4;
+                      switch (_fec_oti.encoding_id) {
+                        case FecScheme::CompactNoCode:
+                          if (hel != 4) {
+                            throw "Invalid length for EXT_FTI header extension for Compact No Code FEC scheme";
+                          }
+                          _fec_oti.transfer_length = (uint64_t)(ntohs(*(uint16_t*)hdr_ptr)) << 32;
+                          hdr_ptr += 2;
+                          _fec_oti.transfer_length |= (uint64_t)(ntohl(*(uint32_t*)hdr_ptr));
+                          hdr_ptr += 4;
+                          hdr_ptr += 2; // reserved
+                          _fec_oti.encoding_symbol_length = ntohs(*(uint16_t*)hdr_ptr);
+                          hdr_ptr += 2;
+                          _fec_oti.max_source_block_length = ntohl(*(uint32_t*)hdr_ptr);
+                          hdr_ptr += 4;
+                          break;
+                        case FecScheme::Raptor:
+                          //TODO
```

Comment:
What is missing here?

Reply:
So the FLUTE specification requires the receiver be able to process FEC parameters from **both** the FDT and the EXT_FTI header extension, whichever one the transmitter chooses to use. However our implementation only ever uses the FDT to communicate these parameters, so we didnt implement the use of EXT_FTI extension.